feature(broker) add capability to sent events to a socket stream

cityhunteur opened the pull request:
This allows the Vega core to stream events to the DataNode, i.e. a remote broker.

TODOs
- [x] Improve logging
- [x] Graceful termination of routines

close #3819

Commit:
feature(broker) add capability to sent events to a socket stream

This allows the Vega core to stream events to the DataNode, i.e.
a remote broker.

jeremyletang reviewed (commented):
just a first pass at the review.

Commit:
feature(broker): add IP config for socket endpoint

This also renames the socket client.

Commit:
feature(broker) refactor error handling and event marshaling

Commit:
feature(broker): add Market to MarketEvent proto

Commit:
feature(broker): refactor socket client dial logic

Commit:
feature(broker): update default socket config

Commit:
feature(broker): refactor broker calls to socket client

Commit:
feature(broker): refactor socket client

This changes the client to a buffered channel to receive events from
the broker and also to send the events over the socket.

The client also tries to reconnect in case of a connection failure.

Commit:
fix(events): update bus event type for MarketCreated event

Commit:
fix(events): update bus event type for MarketUpdated event

Commit:
refactor(cmd/verify): use a noop broker

Commit:
fix(broker): mock for socket client

Commit:
refactor(api): update to use %w for wrapping err

Commit:
refactor(broker): close socket only when socket client is enabled

Commit:
refactor(broker): add socket client configuration

This also improves error handling and closing of channels.

Commit:
refactor(subscribers): fix stubs that implements the Event interface

This also fixes some linting issues.

Commit:
refactor(subscribers): resolve pr comments

karlem committed:
update broker

karlem committed:
terminate on too many events

karlem committed:
add broker streaming tests

karlem committed:
fix comments

karlem committed:
use noop broker and fix compilation error

jeremyletang reviewed (dismissed):
@EVODelavega @peterbarrow anymore comments on this one?

karlem committed:
add this annoying thing
